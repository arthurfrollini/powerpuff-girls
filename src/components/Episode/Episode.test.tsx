import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import { Episode } from ".";

import renderer from "react-test-renderer";

describe("Episode component", () => {
  it("renders correctly", () => {
    const { getByText } = render(
      <Router>
        <Episode name="episode" id="1" episodeNumber={1} />
      </Router>
    );

    expect(getByText("episode")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const tree = renderer
      .create(
        <Router>
          <Episode name="episode" id="1" episodeNumber={1} />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
