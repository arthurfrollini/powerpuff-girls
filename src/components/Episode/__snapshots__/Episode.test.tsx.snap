// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episode component matches snapshot 1`] = `
<a
  className="sc-bdvvtL egPeii"
  href="/episode/1"
  onClick={[Function]}
>
  <span
    className="sc-gsDKAQ ksHaVr"
  >
    2
  </span>
  <strong
    className="sc-dkPtRN"
  >
    episode
  </strong>
</a>
`;
